system/jlib/jptree.cpp

```diff
@@ -6407,11 +6407,24 @@ class CJSONReaderBase : public CommonReaderBase<X>
             if (!isdigit(nextChar))
                 error("Bad value");
             type = elementTypeInteger;
-            while (isdigit(nextChar) || '.'==nextChar)
+            bool exponent = false;
+            while (isdigit(nextChar) || '.'==nextChar || 'e'==nextChar || 'E'==nextChar)
             {
+                if ('e'==nextChar || 'E'==nextChar)
+                {
+                    if (exponent)
+                        error("Bad value");
+                    exponent=true;
+                    value.append(nextChar);
+                    readNext();
+                    if ('-'==nextChar)
+                        type=elementTypeReal;
```

Comment:
I don't think it makes a difference either way, but my thinking was that with a positive exponent, and without a decimal point it could be thought of as a big int rather than a real.  But in the end numeric element types are handled the same way.